Runtime/Extensions/GuidBasedReferenceExtensions.cs

```diff
@@ -0,0 +1,52 @@
+using System.Collections.Generic;
+using Object = UnityEngine.Object;
+
+namespace RoyTheunissen.AssetPalette
+{
+    public static class GuidBasedReferenceExtensions
```

Comment:
I think this is cool, but it's a bit unnecessary to be in an Extension, since this probably only needs to exist in one release, and can be removed later on. 

We just need to explain to users that keep compatibility with the new system, they need to upgrade passing through this version.

Reply:
Not really. 

What we do is, release 0.3.3 with this. 

Then on 0.3.4 we can already remove and add a warning, saying that anyone upgrading from the previous version, should upgrade to 0.3.3 before upgrading to any other version ahead. 

But honestly, its quite small and I don't mind keeping either.